packages/medusa/src/api/routes/admin/inventory-items/utils/join-levels.ts

```diff
@@ -0,0 +1,78 @@
+import { IInventoryService } from "../../../../../interfaces"
+import {
+  InventoryItemDTO,
+  InventoryLevelDTO,
+} from "../../../../../types/inventory"
+
+type LevelWithAvailability = InventoryLevelDTO & {
+  available_quantity: number
+}
+
+export const buildLevelsByInventoryItemId = (
+  inventoryLevels: (LevelWithAvailability | InventoryLevelDTO)[],
+  locationIds: string[]
+) => {
+  return inventoryLevels.reduce((acc, level) => {
+    if (locationIds.length) {
+      if (!locationIds.includes(level.location_id)) {
```

Comment:
question: If the number of location is relatively low then it is fine. otherwise prefer using a map to avoid loop in loop